src/primitives/ray.jl

```diff
@@ -42,3 +42,17 @@ origin(r::Ray) = r.p
 The direction of the ray.
 """
 direction(r::Ray) = r.v
+
+
+
+function Base.in(p::Point, r::Ray)
+	w = norm(r.v)
+	d = evaluate(Euclidean(), p, Line(r(0), r(1)))
+	if d+w ≈ w
+		return sum((p - r.p) .* r.v) >= 0 # return true if point is in ray
```

Comment:
Incorrect code style, please fix it to follow the style of the project.

Also, I understood that you are just repeating the code of point and line and then doing an additional check to see if the point is on the right side of the ray. In this case keep it high-level:

```julia
p \in Line(r(0), r(1)) && # additional check
```

Also, notice that 0 and 1 in Julia are Int. You need to use `zero(T)` and `one(T)` to get the same coordinate type.